vhdbuilder/packer/init-variables.sh

```diff
@@ -44,15 +44,17 @@ echo "CAPTURED_SIG_VERSION set to: ${CAPTURED_SIG_VERSION}"
 echo "Subscription ID: ${SUBSCRIPTION_ID}"
 echo "Service Principal Path: ${SP_JSON}"
 
-if [ -a "${SP_JSON}" ]; then
-	echo "Existing credentials file found."
-	exit 0
-elif [ -z "${CLIENT_ID}" ]; then
-	echo "Service principal not found! Generating one @ ${SP_JSON}"
-	az ad sp create-for-rbac -n aks-images-packer${CREATE_TIME} -o json > ${SP_JSON}
-	CLIENT_ID=$(jq -r .appId ${SP_JSON})
-	CLIENT_SECRET=$(jq -r .password ${SP_JSON})
-	TENANT_ID=$(jq -r .tenant ${SP_JSON})
+if [ "$MODE" == "linuxVhdMode" ]; then
```

Comment:
Can we assure that we have only two modes, linuxVhdMode and windowsVhdMode? If not, I think that it is better to use `"$MODE" != "windowsVhdMode"`

Reply:
@cameronmeissner, it seems like that we only use `linuxVhdMode` and `windowsVhdMode` now and we can remove those legacy code for `default` and `sigMode`. Could you help to confirm it?